### PR TITLE
feat: Add new ptn modules `avm/ptn/azd/aks-automatic-cluster`

### LINKED ISSUE #1426: [Module Proposal]: `avm/ptn/azd/aks-automatic-cluster`
### Check for previous/existing GitHub issues/module proposals

- [X] I have checked for previous/existing GitHub issues/module proposals.

### Check this module doesn't already exist in the module indexes

- [X] I have checked for that this module doesn't already exist in the module indexes.

### Bicep or Terraform?

Bicep

### Module Classification?

Pattern Module

### Module Name

avm/ptn/azd/aks-automatic-cluster

### Module Details

Please add a Pattern Module named `avm/ptn/azd/aks-automatic-cluster`, this module will create the following three resources:

1. Azure Kubernetes Service (AKS) Automatic Cluster.
2. Azure Container Registry (ACR).
3. Key Vault.

For details, please refer to the related PR:
https://github.com/Azure/azure-dev/pull/4069

CC: @jongio, @wbreza 

### Do you want to be the owner of this module?

No

### Module Owner's GitHub Username (handle)

_No response_

### (Optional) Secondary Module Owner's GitHub Username (handle)

_No response_

### PR DESCRIPTION
**Pull request by zedy-wj**
## Description

Fixes https://github.com/Azure/Azure-Verified-Modules/issues/1426

## Pipeline Reference

| Pipeline |
| -------- |
|[avm.ptn.azd.aks-automatic-cluster](https://github.com/zedy-wj/bicep-registry-modules/actions/workflows/avm.ptn.azd.aks-automatic-cluster.yml)|

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: .github/CODEOWNERS
Comment (AlexanderSehr) [inline .github/CODEOWNERS:1]: Please note, the `main.json` templates **must** be re-build with the latest Bicep version `0.31.34` as this version (running on the agents) renders a different `main.json` file then previous versions. 

The steps to execute would be
- Upgrade your Bicep CLI version to `0.31.34` (e.g. via `choco upgrade bicep` if choco is used)
- Re-run the `Set-AVMModule` script on your module. E.g. via `Set-AVMModule -ModuleFolderPath '.\avm\res\automation\automation-account\' -SkipFileAndFolderSetup -SkipReadMe -Recurse`
- Push the changes
- Re-run the tests to ensure everything works. 

If the static tests continue to fail for you, please note that there is currently presumably another new bug which affects some of our modules ([ref](https://github.com/Azure/bicep/issues/15517)).

**Comment by jtracey93** [comment]
Hey @zedy-wj ,

Firstly, thanks for your work on this PR!

We have made some changes to the AVM CI, detailed below, which means we need you to update your fork to pull in these latest changes and re-run your tests to show they still are passing prior to approving and merging this PR, as we don't and it fails once merged the publishing of your module will fail and will be blocked going forward until the test pass again via additional PRs.

### Changes to CI That Have Been Made That You Need To Take Action On

- https://github.com/Azure/bicep-registry-modules/pull/3745
    - Adds WAF Security PSRule Checks using this custom baseline of rules ([`/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/cb-waf-security.Rule.yaml`](https://github.com/Azure/bicep-registry-modules/blob/main/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/cb-waf-security.Rule.yaml))
    - **Action Required By You:**
        - Ingest/pull-in these changes into your PR branch and re-run your tests to ensure they pass
            - If any issues with the rules and you think it should apply, please ping/reach out to the core team

Any questions reach out to the AVM Core Team by tagging us in your PR here or internally via Teams

Thanks

Jack (AVM Core Team)

File: .github/CODEOWNERS
Comment (zedy-wj) [reply .github/CODEOWNERS:1]: Done.

**Comment by zedy-wj** [comment]
@AlexanderSehr - This PR is ready now, please review it, thanks!

File: avm/ptn/azd/aks-automatic-cluster/tests/e2e/defaults/main.test.bicep
Comment (AlexanderSehr) [inline avm/ptn/azd/aks-automatic-cluster/tests/e2e/defaults/main.test.bicep:4]: Not sure - but is this description accurate? I want to say yes because I've seen the word `auto` somwhere, but wanted to double-check

Comment (zedy-wj) [reply avm/ptn/azd/aks-automatic-cluster/tests/e2e/defaults/main.test.bicep:4]: Yes, it is indeed a test of an AKS cluster. I'm guessing you're seeing "another" `auto` [here](https://github.com/Azure/bicep-registry-modules/blob/main/avm/res/container-service/managed-cluster/tests/e2e/automatic/main.test.bicep#L5). They have some overlap, but we want to implement a template in the `ptn` module based on the default AKS cluster.

For more details, you can get from this [comment](https://github.com/Azure/bicep-registry-modules/pull/3104#discussion_r1769021485).

Comment (AlexanderSehr) [reply avm/ptn/azd/aks-automatic-cluster/tests/e2e/defaults/main.test.bicep:4]: Thanks for elaborating :) Resolving

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**
